Configurable Logger

njbrake opened the pull request:
Users are going to want to configure/disable the any_agent logging. This PR adds this ability, as well as a generic pass through to the rich logger.

daavoo commented on tests/conftest.py:
```diff
@@ -1,32 +1,29 @@
+from any_agent.logging import setup_logger
 
 
-@pytest.fixture(autouse=True)
```
@njbrake this was not used for the logger but to avoid tracing output (the rich console usage) during test runs.

I don't think this change preserves the previously intended behavior

njbrake replied:
So I did test it and with my changes it was correctly suppressing console output when `-s` was removed, and adding it in when `-s` was added. Was it supposed to do something else? (That's what I thought it was supposed to do)

daavoo replied:
I don't understand what changed since I added the fixture but it seems that even if we remove your fixture the rich console gets correctly disabled. Maybe at some rich/pytest version this got fixed

njbrake replied:
Noice. So I guess maybe we can remove the fixture completely. I'll test.

daavoo reviewed (commented):
How would users modify the default behavior? 
Have you tested it? Can we test it and/or add it to the docs?

njbrake commented:
@daavoo  ok I think we're in a better place now:
* added docs for logging
* cleaned up rich console log handler in tests
* added unit tests for console
